src/lib.rs

```diff
@@ -41,6 +41,12 @@ pub struct Suggestion {
     pub replacement: String,
 }
 
+impl Suggestion {
+    pub fn is_use_suggestion(&self) -> bool {
+        self.replacement.starts_with("use ") && self.replacement.ends_with(";\n")
```

Comment:
Can we get a more stable detection here?

Rust's errors have codes that are present in the JSON output like this:

```js
{
    message: {
        children: [{
            code: null, level: "help", message: "", suggested_replacement: "..."
        }],
        code: {
            code: "E0425", explanation: "..."
        },
        level: "error",
        message: "...",
        spans: []
    }
}
```

So, here, we can probably look for the `message.code.code == E0425`. This might require adjusting the `Suggestions` struct.

- - -

For future extensions to other lints: Most clippy lint messages start with something like

```json
"children": [
    {
        "children": [],
        "code": null,
        "level": "note",
        "message": "#[warn(needless_borrow)] on by default",
        "rendered": null,
        "spans": []
    },
    {
        "children": [],
        "code": null,
        "level": "help",
        "message": "for further information visit https://github.com/Manishearth/rust-clippy/wiki#needless_borrow",
        "rendered": null,
        "spans": []
    }
]
```

Parsing the `"#[warn(needless_borrow)] on by default"` message seems a bit less fragile than "everything that begins with use and ends with a semicolon". It would be even more awesome to get the actual lint name as structured data (maybe even lint group as well as the child lint that actually created the diagnostic message), but that's something that needs changes in rustc (and some discussion, maybe even an RFC).

Reply:
What do you think about placing the lint name into the `code` field?

Reply:
As a string or something like `"code": { "code": null, "lint": "needless_borrow" }`?

Reply:
I'd have thought `"code": { "code": "needless_borrow", "explanation": "..." }`

Reply:
Sure. I have no idea if error codes and lints ever overlap (i.e., if a lint can have an error code or an error with a code a lint name)